src/index.js

```diff
@@ -1 +1,90 @@
 'use strict';
+
+const http = require('http');
+const fs = require('fs');
+const path = require('path');
+const formidable = require('formidable');
+
+const PORT = process.env.PORT || 3001;
+
+const server = http.Server();
+
+server.on('request', (req, res) => {
+  if (req.method === 'POST' && req.url === '/submit') {
+    const form = new formidable.IncomingForm();
+
+    form.parse(req, (err, fields, files) => {
+      if (err) {
+        res.status = 400;
+        res.end(JSON.stringify(err));
+      }
+
+      const { date, title, amount } = fields;
+
+      const data = {
+        date: date[0],
+        title: title[0],
+        amount: amount[0],
+      };
+
+      const jsonFilePath = path.join(__dirname, 'output.json');
+      let allData;
+
+      try {
+        allData = JSON.parse(fs.readFileSync(jsonFilePath, 'utf-8', (error) => {
+          if (error) {
+            // eslint-disable-next-line no-console
+            console.log(error);
+          }
+        }));
+      } catch (error) {
+        allData = [];
+      }
```

Comment:
You use two error handlers for the same operation. It's redundant. Move the logic in one place